sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/serializer/TypeReference.java

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.experimental.serializer;
+
+import com.azure.core.util.logging.ClientLogger;
+
+import java.lang.reflect.ParameterizedType;
+
+/**
+ * This class represents a generic Java type, retaining information about generics.
+ *
+ * @param <T> The type being represented.
+ */
+public abstract class TypeReference<T> {
```

Comment:
Why is this abstract?

Reply:
This is abstract to leverage functionality of the Java type system. If this we a concrete class we would have access to the `T` type as the generic super class would be `Object`. But if this is abstract and the only way to use it is with anonymous, or concrete, sub classes the generic super type would be the `T` value a user expected.